### PR TITLE
Tweak getter function made throwable

### PR DESCRIPTION
**Pull request by saniazafar**
## Description
- The getter function of Tweak is made throwable

## Motivation and Context
 - With the addition of these changes #63 we now have multiple reasons where Tweak can be a nil value.
 - To better capture the error the getter is made throwable.

## How Has This Been Tested?
 - Tested the example app and updated the Unit tests.

## Screenshots (if appropriate):
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by albertodebortoli — COMMENTED** [review]
Beautifully done @saniazafar! 👏 
Just a couple of comments.

**Comment by saniazafar** [comment]
Thanks @albertodebortoli. I have addressed your comments.

**Review by albertodebortoli — APPROVED** [review]

File: Example/JustTweak/Accessors/TweakAccessor.swift [hunk 1]
```diff
@@ -87,18 +87,18 @@ class TweakAccessor {
     // MARK: - Via TweakManager
     
     var canShowYellowView: Bool {
-        return tweakManager.tweakWith(feature: Features.uiCustomization,
-                                      variable: Variables.displayYellowView)?.boolValue ?? false
+        return (try? tweakManager.tweakWith(feature: Features.uiCustomization,
+                                      variable: Variables.displayYellowView))?.boolValue ?? false
```
Comment (dchakarov) [inline Example/JustTweak/Accessors/TweakAccessor.swift:91]: You have to reindent this (and below)

File: Example/Tests/Core/LocalTweakProviderTests.swift [hunk 2]
```diff
@@ -32,7 +32,7 @@ class LocalTweakProviderTests: XCTestCase {
                                  value: true,
                                  title: "Display Red View",
                                  group: "UI Customization")
-        XCTAssertEqual(redViewTweak, tweakProvider.tweakWith(feature: Features.uiCustomization,
+        XCTAssertEqual(redViewTweak, try tweakProvider.tweakWith(feature: Features.uiCustomization,
                                                              variable: Variables.displayRedView))
```
Comment (dchakarov) [inline Example/Tests/Core/LocalTweakProviderTests.swift:36]: You have to reindent this (and below)

File: Example/Tests/Core/UserDefaultsTweakProviderTests.swift [hunk 3]
```diff
@@ -31,45 +31,45 @@ class UserDefaultsTweakProviderTests: XCTestCase {
...
+        let tweak = try? userDefaultsTweakProvider.tweakWith(feature: Features.uiCustomization, variable: Variables.displayRedView)
         XCTAssertNil(tweak)
     }
     
     func testUpdatesValueForTweak_withBool() {
         userDefaultsTweakProvider.set(true, feature: "feature_1", variable: "variable_1")
-        let tweak = userDefaultsTweakProvider.tweakWith(feature: "feature_1", variable: "variable_1")
-        XCTAssertTrue(tweak!.value == true)
+        let tweak = try! userDefaultsTweakProvider.tweakWith(feature: "feature_1", variable: "variable_1")
+        XCTAssertTrue(tweak.value == true)
```
Comment (dchakarov) [inline Example/Tests/Core/UserDefaultsTweakProviderTests.swift:48]: You can use `XCTUnwrap` instead
```suggestion
        let tweak = try XCTUnwrap(userDefaultsTweakProvider.tweakWith(feature: "feature_1", variable: "variable_1"))
        XCTAssertTrue(tweak.value)
```

File: README.md [hunk 4]
```diff
@@ -131,19 +131,31 @@ if enabled {
...
+Or with do-catch
+```swift
+// check for a tweak value
+do {
+    let tweak = try tweakManager.tweakWith(feature: "some_feature", variable: "some_flag")
+    // tweak was found in some tweak provider, use tweak.value
+    return tweak    
+} catch let error as TweakError {
+    // tweak was not found in any tweak provider
+    //handle error
```
Comment (dchakarov) [inline README.md:156]:
```suggestion
    switch error {
    case .notFound: () // "Feature or variable is not found"
    case .notSupported: () // "Variable type is not supported"
    case .decryptionClosureNotProvided: () // "Value is encrypted but there's no decryption closure provided"
    }
```

File: README.md [hunk 5]
```diff
@@ -131,19 +131,31 @@ if enabled {
...
+}
+```
```
Comment (dchakarov) [inline README.md:158]: You might need to add a default catch (e.g. `} catch let error { print(error.localizedDescription) }`) to satisfy the compiler.

**Review by dchakarov — COMMENTED** [review]
Nice work! I left a few minor suggestions

**Comment by saniazafar** [comment]
Thanks for the suggestions @dchakarov 👍

**Review by dchakarov — APPROVED** [review]

**Pull request merged**